analyzer/windows/modules/auxiliary/html_scrap.py

```diff
@@ -0,0 +1,112 @@
+from lib.core.config import Config
+
+log = logging.getLogger(__name__)
+
+HAVE_SELENIUM = False
+
+try:
+    from selenium import webdriver
+    from selenium.webdriver.firefox.options import Options
+
+    HAVE_SELENIUM = True
+except Exception as ex:
+    log.error(ex)
+
+__author__ = "Jonathan Abrahamy"
+__email__ = "jonathan@intezer.com"
+__version__ = "0.0.1"
+__date__ = "03MAY2022"
+
+
+class HtmlScrap(Thread, Auxiliary):
+    def __init__(self, options=None, config=None):
+        Thread.__init__(self)
+        Auxiliary.__init__(self, options, config)
+        self.config = Config(cfg='analysis.conf')
+
+        self.enabled = True
+        if 'html_scrap' in options:
+            self.enabled = options['html_scrap']
+
+        self.driver_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..', '..', 'bin',
+                                        'geckodriver.exe')
+
+    @staticmethod
+    def upload_to_htmlscrap_folder(file_name: str, content: bytes):
+        tmpio = BytesIO(content)
+
+        nf = NetlogFile()
+        nf.init(f'htmlscrap/{file_name}')
+        # now upload to host from the StringIO
+        for chunk in tmpio:
+            nf.sock.send(chunk)
+        nf.close()
+
+    def scrap_html(self):
+        if not HAVE_SELENIUM:
+            log.debug('Selenium not installed on machine, not scraping', self.driver_path)
+            return
+
+        if not os.path.isfile(self.driver_path):
+            log.debug('Web driver not found in path %s, not scraping', self.driver_path)
+            return
+
+        if not hasattr(self.config, "category") or self.config.category != 'file':
+            log.debug('Category is not file, not scraping', self.config.category)
+            return
+
+        if not hasattr(self.config, "file_type") or 'HTML' not in self.config.file_type:
+            log.debug('File is not html, not scraping', self.config.category)
+            return
+
+        try:
+            file_path = os.path.join(os.environ["TEMP"] + os.sep, str(self.config.file_name))
+
+            opts = Options()
+            opts.headless = True
+
+            # Set fake proxy to prevent internet connectivity
+            opts.set_preference('network.proxy.type', 1)
+            opts.set_preference('network.proxy.socks', '127.0.0.1')
+            opts.set_preference('network.proxy.socks_port', 9050)
+            opts.set_preference('network.proxy.socks_remote_dns', False)
+
+            browser = webdriver.Firefox(options=opts, executable_path=self.driver_path)
+
+            sample_url = 'file:///{}'.format(os.path.abspath(file_path))
+            browser.get(sample_url)
+            time.sleep(3)
+
+            log.debug('Starting upload')
+            self.upload_to_htmlscrap_folder('scrap.dump', browser.page_source.encode())
+
+            if not browser.current_url.startswith('file://'):
+                self.upload_to_htmlscrap_folder('scrap.dump', browser.current_url.encode())
```

Comment:
`browser.current_url.encode()` - so we would add the redirect url? if so, is there more elegant way?